### PR TITLE
Add CommandConverter and CogConverter + add usage in Core

### PR DESCRIPTION
**Pull request by phenom4n4n**
Adds a command and cog converter to the `commands.converter` module, which would be useful in several core commands and various cogs.

### CODE REVIEW

**Comment by phenom4n4n** [comment]
This PR is ready for review.

File: redbot/core/commands/converter.py [hunk 1]
```diff
@@ -416,3 +418,33 @@ def __class_getitem__(cls, k):
...
+
+if TYPE_CHECKING:
+    CommandConverter = dpy_commands.Command
+    CogConverter = dpy_commands.Cog
+else:
+
+    class CommandConverter(dpy_commands.Converter):
+        """
+        Returns a `Command` object if the passed argument matches a command on the bot.
+        """
```
Comment (Jackenmen) [inline redbot/core/commands/converter.py]:
```suggestion
        """Converts a command name to the matching `Command` object."""
```

File: redbot/core/commands/converter.py [hunk 2]
```diff
@@ -416,3 +418,33 @@ def __class_getitem__(cls, k):
...
+            arg = argument.strip()
+            command = ctx.bot.get_command(arg)
+            if not command:
+                raise BadArgument(_('Command "{arg}" not found.').format(arg=arg))
+            return command
+
+    class CogConverter(dpy_commands.Converter):
+        """
+        Returns a `Cog` object if the passed argument matches a cog name on the bot.
+        """
```
Comment (Jackenmen) [inline redbot/core/commands/converter.py]:
```suggestion
        """Converts a cog name to the matching `Cog` object."""
```

File: redbot/core/core_commands.py [hunk 3]
```diff
@@ -1151,15 +1154,9 @@ async def embedset_showsettings(self, ctx: commands.Context, command_name: str =
         **Arguments:**
             - `[command_name]` - Checks this command for command specific embed settings.
```
Comment (Jackenmen) [inline redbot/core/core_commands.py]: Argument name was changed: 
```suggestion
            - `[command]` - Checks this command for command specific embed settings.
```
This also needs equivalent change in the docs.

File: redbot/core/core_commands.py [hunk 4]
```diff
@@ -4060,7 +4045,7 @@ async def command_manager(self, ctx: commands.Context):
 
     @checks.is_owner()
     @command_manager.command(name="defaultdisablecog")
-    async def command_default_disable_cog(self, ctx: commands.Context, *, cogname: str):
+    async def command_default_disable_cog(self, ctx: commands.Context, *, cogname: CogConverter):
```
Comment (Jackenmen) [inline redbot/core/core_commands.py]: As with command_name->command, this should also be renamed from cogname to cog:
```suggestion
    async def command_default_disable_cog(self, ctx: commands.Context, *, cog: CogConverter):
```
You will need to update the docstring and docs as well.

File: redbot/core/core_commands.py [hunk 5]
```diff
@@ -4075,17 +4060,16 @@ async def command_default_disable_cog(self, ctx: commands.Context, *, cogname: s
...
+        cogname = cog.qualified_name
         if isinstance(cog, commands.commands._RuleDropper):
             return await ctx.send(_("You can't disable this cog by default."))
         await self.bot._disabled_cog_cache.default_disable(cogname)
         await ctx.send(_("{cogname} has been set as disabled by default.").format(cogname=cogname))
 
     @checks.is_owner()
     @command_manager.command(name="defaultenablecog")
-    async def command_default_enable_cog(self, ctx: commands.Context, *, cogname: str):
+    async def command_default_enable_cog(self, ctx: commands.Context, *, cogname: CogConverter):
```
Comment (Jackenmen) [inline redbot/core/core_commands.py]: As with command_name->command, this should also be renamed from cogname to cog:
```suggestion
    async def command_default_enable_cog(self, ctx: commands.Context, *, cog: CogConverter):
```
You will need to update the docstring and docs as well.

File: redbot/core/core_commands.py [hunk 6]
```diff
@@ -4100,15 +4084,14 @@ async def command_default_enable_cog(self, ctx: commands.Context, *, cogname: st
...
-            return await ctx.send(_("Cog with the given name doesn't exist."))
+        cog = cogname
+        cogname = cog.qualified_name
         await self.bot._disabled_cog_cache.default_enable(cogname)
         await ctx.send(_("{cogname} has been set as enabled by default.").format(cogname=cogname))
 
     @commands.guild_only()
     @command_manager.command(name="disablecog")
-    async def command_disable_cog(self, ctx: commands.Context, *, cogname: str):
+    async def command_disable_cog(self, ctx: commands.Context, *, cogname: CogConverter):
```
Comment (Jackenmen) [inline redbot/core/core_commands.py]: As with command_name->command, this should also be renamed from cogname to cog:
```suggestion
    async def command_disable_cog(self, ctx: commands.Context, *, cog: CogConverter):
```
You will need to update the docstring and docs as well.

File: redbot/core/core_commands.py [hunk 7]
```diff
@@ -4134,7 +4116,7 @@ async def command_disable_cog(self, ctx: commands.Context, *, cogname: str):
 
     @commands.guild_only()
     @command_manager.command(name="enablecog")
-    async def command_enable_cog(self, ctx: commands.Context, *, cogname: str):
+    async def command_enable_cog(self, ctx: commands.Context, *, cogname: CogConverter):
```
Comment (Jackenmen) [inline redbot/core/core_commands.py]: As with command_name->command, this should also be renamed from cogname to cog:
```suggestion
    async def command_enable_cog(self, ctx: commands.Context, *, cog: CogConverter):
```
You will need to update the docstring and docs as well.

File: redbot/core/core_commands.py [hunk 8]
```diff
@@ -4276,39 +4260,32 @@ async def command_disable_global(self, ctx: commands.Context, *, command: str):
...
 
-        if isinstance(command_obj, commands.commands._RuleDropper):
+        if isinstance(command, commands.commands._RuleDropper):
             await ctx.send(
                 _("This command is designated as being always available and cannot be disabled.")
             )
             return
 
         async with ctx.bot._config.disabled_commands() as disabled_commands:
             if command not in disabled_commands:
```
Comment (Jackenmen) [inline redbot/core/core_commands.py]: `command` used to be a string so this needs to be updated appropriately:
```suggestion
            if command.qualified_name not in disabled_commands:
```

File: redbot/core/core_commands.py [hunk 9]
```diff
@@ -4319,42 +4296,35 @@ async def command_disable_guild(self, ctx: commands.Context, *, command: str):
...
             )
             return
 
-        if command_obj.requires.privilege_level > await PrivilegeLevel.from_ctx(ctx):
+        if command.requires.privilege_level > await PrivilegeLevel.from_ctx(ctx):
             await ctx.send(_("You are not allowed to disable that command."))
             return
 
         async with ctx.bot._config.guild(ctx.guild).disabled_commands() as disabled_commands:
             if command not in disabled_commands:
```
Comment (Jackenmen) [inline redbot/core/core_commands.py]: `command` used to be a string so this needs to be updated appropriately:
```suggestion
            if command.qualified_name not in disabled_commands:
```

File: redbot/core/core_commands.py [hunk 10]
```diff
@@ -1249,7 +1246,7 @@ async def embedset_guild(self, ctx: commands.Context, enabled: bool = None):
     @checks.guildowner_or_permissions(administrator=True)
     @embedset.group(name="command", invoke_without_command=True)
     async def embedset_command(
-        self, ctx: commands.Context, command_name: str, enabled: bool = None
+        self, ctx: commands.Context, command: CommandConverter, enabled: bool = None
```
Comment (Jackenmen) [inline redbot/core/core_commands.py:1250]: The argument name was changed so this also needs equivalent change in the docs.

File: redbot/core/core_commands.py [hunk 11]
```diff
@@ -1289,7 +1286,7 @@ def _check_if_command_requires_embed_links(self, command_obj: commands.Command)
     @commands.is_owner()
     @embedset_command.command(name="global")
     async def embedset_command_global(
-        self, ctx: commands.Context, command_name: str, enabled: bool = None
+        self, ctx: commands.Context, command: CommandConverter, enabled: bool = None
```
Comment (Jackenmen) [inline redbot/core/core_commands.py:1290]: The argument name was changed so this also needs equivalent change in the docs.

File: redbot/core/core_commands.py [hunk 12]
```diff
@@ -1340,7 +1331,7 @@ async def embedset_command_global(
     @commands.guild_only()
     @embedset_command.command(name="server", aliases=["guild"])
     async def embedset_command_guild(
-        self, ctx: commands.GuildContext, command_name: str, enabled: bool = None
+        self, ctx: commands.GuildContext, command: CommandConverter, enabled: bool = None
```
Comment (Jackenmen) [inline redbot/core/core_commands.py:1335]: The argument name was changed so this also needs equivalent change in the docs.

**Review by Jackenmen — CHANGES_REQUESTED** [review]

**Review by Jackenmen — APPROVED** [review]
Looks good to me, thanks for the PR!

**Pull request merged**